lib/pure/uri.nim

```diff
@@ -311,6 +356,46 @@ proc `$`*(u: Uri): string =
     result.add("#")
     result.add(u.anchor)
 
+proc parseQuery*(query: string): OrderedTableRef[string, string] = 
+  ## parses the query part of an uri.
+  var start = 0
+  var key, val = ""
+  var queryBlock = ""
+  if query == "":
+    return
+  result = newOrderedTable[string, string]()
+  for i in 0..query.len:
+    if query[i] == '&' or i == query.len:
+      start = i
+      var inval = false
+      for each in queryBlock:
```

Comment:
Try simplifying this if you can. I think this can be done with a single while loop instead of two for loops.